z230-pt1.md

```diff
@@ -0,0 +1,537 @@
+Python 0-30 for Scientists
+==========================
+
+This is an outline of the steps needed to teach Python 0-30.  The plan is to teach
+Python in an example-based way demonstrating:
+
+1. How to write Python code
+1. Python best-practices
+1. Coding with git & GitHub
+1. Package management with Conda
+
+with each being of equal importance.  This means that we intend to:
+
+1. Use "the right" data structures for the intended task, without needed to depend upon external 3rd-party packages.
+1. Commit early and commit often, so that each "lesson" is represented by a single git commit.
+1. End the instruction with an installable package and demonstrate the usefulness of such packages.
+
+Requirements & Assumptions
+--------------------------
+
+- Assumes a linux/mac OS (What about Windows?)
+- Requires conda
+
+Outline - Part 1 - First Python Script
+--------------------------------------
+
+1. [bash] Create a directory:
+
+   ```bash
+   $ mkdir mysci
+   ```
+   
+1. [bash] Go into the directory:
+
+   ```bash
+   $ cd mysci
+   ```
+   
+1. [conda] Create a virtual environment for this project:
+
+   ```bash
+   $ conda create --name mysci python
+   ```
+   
+   **NOTE:** Introduce conda and conda environments
+
+1. [git] Make the directory a git repository:
+
+   ```bash
+   $ git init .
+   ```
+   
+   **NOTE:** Introduce Git
+   
+1. [bash] Create a data directory:
+
+   ```bash
+   $ mkdir data
+   ```
+   
+1. [bash] Go into the data directory:
+
+   ```bash
+   $ cd data
+   ```
+   
+1. [bash] Download sample data:
+
+   ```bash
+   $ curl -O https://sundowner.colorado.edu/weather/atoc8/wxobs20170821.txt
+   ```
+   
+   **NOTE:** A note about the CU Boulder weather station
```

Comment:
This weather station is a Davis Instruments wireless Vantage Pro2 located on the CU-Boulder east campus at the SEEC building (40.01 N, 05.24 W, 5250 ft elevation). The station is monitored by the Atmospheric and Oceanic Sciences (ATOC) department and is part of the larger University of Colorado ATOC Weather Network.


---------------
But perhaps we want to use data provided by NCAR? There is a station here at the Mesa Lab.
https://archive.eol.ucar.edu/cgi-bin/weather.cgi?site=ml&period=5-minute&units=english&&fields=tdry&fields=rh&fields=cpres0&fields=wspd&fields=wdir&fields=raina